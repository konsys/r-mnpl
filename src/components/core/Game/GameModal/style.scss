.gameModal {
  width: 100%;
  min-width: 180px;
  max-width: 1000px;
  min-height: 170px;
  overflow: hidden;
}
