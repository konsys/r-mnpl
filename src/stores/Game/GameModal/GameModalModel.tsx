import { GameDomain } from "stores/Board/BoardDomain";

const GameModalDomain = GameDomain.domain("GameModalDomain");
export interface IGameModal {
  title: string;
  text: string;
}
export const openGameModal = GameModalDomain.event<IGameModal>();
export const closeGameModal = GameModalDomain.event();
export const gameModal$ = GameModalDomain.store<IGameModal>({
  text: "",
  title: "",
})
  .on(openGameModal, (_, payload) => payload)
  .reset(closeGameModal);
