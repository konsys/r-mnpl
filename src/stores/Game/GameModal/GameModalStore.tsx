import { GameDomain } from "../UserStore";

const GameModalDomain = GameDomain.domain("ModalDomain");
export interface IGameModal {
  open: boolean;
  title: string;
  text: string;
}
export const openGameModal = GameModalDomain.event<IGameModal>();
export const closeGameModal = GameModalDomain.event();
export const gameModalStore = GameModalDomain.store<IGameModal>({
  open: false,
  text: "",
  title: "",
})
  .on(openGameModal, (_, payload) => payload)
  .reset(closeGameModal);
