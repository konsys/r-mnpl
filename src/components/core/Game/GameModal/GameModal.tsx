import "./style.scss";

import {
  Button,
  Dialog,
  DialogContent,
  Grid,
  Typography,
} from "@material-ui/core";
import {
  closeGameModal,
  gameModalStore,
} from "stores/Game/GameModal/GameModalStore";

import React from "react";
import { useStore } from "effector-react";
import { useTranslation } from "react-i18next";

export default function GameModal() {
  const modal = useStore(gameModalStore);
  const { t } = useTranslation();
  return (
    <Dialog open={modal.open} onClose={() => null}>
      <DialogContent className={"gameModal"}>
        <Grid
          container
          spacing={2}
          direction="column"
          justify="center"
          alignContent="center"
          alignItems="center"
        >
          <Grid item>
            <Typography variant="h6">{t(modal.title)}</Typography>
          </Grid>
          <Grid item>
            <Typography variant="body1">{t(modal.text)}</Typography>
          </Grid>
          <Grid item>
            <Button
              onClick={() => {
                closeGameModal();
              }}
              color="default"
              variant="contained"
            >
              {t("Ok")}
            </Button>
          </Grid>{" "}
        </Grid>
      </DialogContent>
    </Dialog>
  );
}
