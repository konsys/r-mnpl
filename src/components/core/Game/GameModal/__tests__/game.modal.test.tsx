import GameModal from "../GameModal";
import React from "react";
import { shallow } from "enzyme";

describe("Buy gallery test", () => {
  it("should render undefined", () => {
    expect(shallow(<GameModal />)).toMatchSnapshot();
  });

  //   it("should render winner name", () => {
  //     expect(
  //       shallow(<GameModal  />)
  //         .find("._winner")
  //         .text()
  //     ).toBe(`Победитель ${testRoomUser.name}`);
  //   });
});
