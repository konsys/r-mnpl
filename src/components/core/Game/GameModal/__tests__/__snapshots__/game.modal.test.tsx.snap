// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buy gallery test should render undefined 1`] = `ShallowWrapper {}`;
